pkg/cloud/vsphere/provisioner/govmomi/utils.go

```diff
@@ -72,6 +67,62 @@ func (pv *Provisioner) GetKubeadmToken(cluster *clusterv1.Cluster) (string, erro
 	return token, err
 }
 
+// The logic for this function comes from various places in kubeadm itself
+func (pv *Provisioner) createKubeadmToken(kubeconfig string) (string, error) {
+	// Create the actual token string
+	token, err := tokenutil.GenerateBootstrapToken()
+	if err != nil || token == "" {
+		return "", fmt.Errorf("unable to create kubeadm token: %s", err)
+	}
+
+	// Create a secret to write back to the cluster
+	tokenID := token[0 : tokenapi.BootstrapTokenIDBytes-1]
+	tokenSecret := token[tokenapi.BootstrapTokenIDBytes:]
+	expiration := time.Now().UTC().Add(constants.KubeadmTokenTtl).Format(time.RFC3339)
+	data := map[string][]byte{
+		tokenapi.BootstrapTokenIDKey:               []byte(tokenID),
+		tokenapi.BootstrapTokenSecretKey:           []byte(tokenSecret),
+		tokenapi.BootstrapTokenExpirationKey:       []byte(expiration),
+		tokenapi.BootstrapTokenUsageAuthentication: []byte("true"),
+		tokenapi.BootstrapTokenUsageSigningKey:     []byte("true"),
+		tokenapi.BootstrapTokenExtraGroupsKey:      []byte("system:bootstrappers:kubeadm:default-node-token"),
+		tokenapi.BootstrapTokenDescriptionKey:      []byte("bootstrap token generated by cluster-api-provider-vsphere"),
+	}
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      tokenutil.BootstrapTokenSecretName(tokenID),
+			Namespace: metav1.NamespaceSystem,
+		},
+		Type: corev1.SecretType(tokenapi.SecretTypeBootstrapToken),
+		Data: data,
+	}
+
+	// Create a client to the target cluster
+	rc, err := clientcmd.RESTConfigFromKubeConfig([]byte(kubeconfig))
+	if err != nil {
+		return "", err
+	}
+
+	clusterclient, err := kubernetes.NewForConfig(rest.AddUserAgent(rc, "cluster-api-provider-vsphere"))
+	if err != nil {
+		glog.Fatalf("Invalid API configuration for kubeconfig: %v", err)
```

Comment:
Why glog.Fatalf here?
This will kill the pod and then keep the actuator in a redeploy loop.
Wouldn't just a `return "", err` suffice here?